service/src/main/kotlin/no/nav/su/se/bakover/service/revurdering/RevurderingServiceImpl.kt

```diff
@@ -67,31 +70,26 @@ internal class RevurderingServiceImpl(
         val sak = sakService.hentSak(sakId).getOrElse {
             return KunneIkkeOppretteRevurdering.FantIkkeSak.left()
         }
-        val tilRevurdering = sak.behandlinger
-            .filterIsInstance(Søknadsbehandling.Iverksatt.Innvilget::class.java)
-            .filter { fraOgMed.between(it.beregning.getPeriode()) }
 
-        val søknadsbehandling: Søknadsbehandling.Iverksatt.Innvilget = when {
-            tilRevurdering.isEmpty() -> return KunneIkkeOppretteRevurdering.FantIngentingSomKanRevurderes.left()
-            tilRevurdering.size > 1 -> return KunneIkkeOppretteRevurdering.KanIkkeRevurderePerioderMedFlereAktiveStønadsperioder.left()
-            revurderingRepo.hentRevurderingForBehandling(tilRevurdering.single().id) != null -> {
-                return KunneIkkeOppretteRevurdering.KanIkkeRevurdereEnPeriodeMedEksisterendeRevurdering.left()
-            }
-            else -> tilRevurdering.single()
-        }
-        val periode = Periode.tryCreate(fraOgMed, søknadsbehandling.beregning.getPeriode().getTilOgMed()).getOrHandle {
+        val tilRevurdering = sak.vedtakListe
+            .filterIsInstance<IVedtakSomGirUtbetaling>()
```

Comment:
Vil vi egentlig revurdere alle vedtak som kan føre til utbetaling? Klage? Anke? Endring? Annet? Jeg hadde følt meg tryggere hvis vi var spesifikke denne gangen og heller generaliserte senere.

Reply:
Denne logikken kommer til å måtte endres etter hvert uansett; for å støtte f.eks. revurdering dersom det kun har vært avslag osv.
Hva tenker du når du sier at vi skal være spesifikke "denne gangen"? At vi sjekker på at det er enten "innvilget søknadsbehandling" eller "innvilget revurdering"? Akkurat her kommer jo dette interfacet til sin rett imo.

Reply:
Jeg tenkte mer i retningen å bruke `.filterIsInstance<InnvilgetStønad>`, heller enn å bruke Vedtak, IVedtak, IVedtakSomGirUtbetaling eller IVedtakSomIkkeGirUtbetaling. Grunnen til det er at i hvert fall ikke jeg har oversikt over fler/alle behandlingene som dukker opp senere som vil arve disse interfacene. Jeg vet heller ikke hvilke Vedtak som skal være lov å revurdere annet enn Søknadsbehandling+Revurdering. Dette er sannsynligvis informasjon som enten finnes eller kan oppdrives raskt fra fag. På grunn av all denne usikkerheten synes jeg vi skal ha strengere enn åpen logikk, slik at vi ikke åpner så lett for å lage bugs etter hvert som vi legger til fler behandlinger.